File: backend/loan-service/Models/ValidationRule.cs
namespace LoanEligibility.API.Models;

public class ValidationRule
{
    public string Field { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty; // 'min', 'max', 'required', 'pattern', 'custom'
    public object? Value { get; set; }
    public string Message { get; set; } = string.Empty;
}

public class ValidationRulesResponse
{
    public List<ValidationRule> Rules { get; set; } = new();
}

File: frontend/models/ValidationRule.ts

export interface ValidationRule {
    field: string;
    type: 'min' | 'max' | 'required' | 'pattern' | 'custom';
    value?: number | string | RegExp;
    message: string;
  }
  
  export interface ValidationRulesResponse {
    rules: ValidationRule[];
  }